components/notification.php

```diff
@@ -0,0 +1,350 @@
+			'component_action'  => $r['action'],
+			'date_notified'     => $r['date'],
+		) );
+
+		// Silent it before it errors.
+		if ( WP_CLI\Utils\get_flag_value( $assoc_args, 'silent' ) ) {
+			return;
+		}
+
+		if ( ! is_numeric( $id ) ) {
+			WP_CLI::error( 'Could not create notification.' );
+		}
+
+		if ( WP_CLI\Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+			WP_CLI::line( $id );
+		} else {
+			WP_CLI::success( sprintf( 'Successfully created new notification (ID #%d)', $id ) );
+		}
+	}
+
+	/**
+	 * Get specific notification.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <notification-id>
+	 * : Identifier for the notification.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 *  ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - ids
+	 *   - json
+	 *   - count
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp notification get 500
+	 *     $ wp bp notification get 56 --format=json
+	 *
+	 * @alias see
+	 */
+	public function get( $args, $assoc_args ) {
+		$notification = bp_notifications_get_notification( $args[0] );
+
+		if ( empty( $notification->id ) ) {
+			WP_CLI::error( 'No notification found by that ID.' );
+		}
+
+		if ( ! is_object( $notification ) ) {
+			WP_CLI::error( 'Could not find the notification.' );
+		}
+
+		$notification_arr = get_object_vars( $notification );
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( $notification_arr );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+		$formatter->display_item( $notification_arr );
+	}
+
+	/**
+	 * Delete a notification.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <notification-id>...
+	 * : ID or IDs of notification to delete.
+	 *
+	 * [--yes]
+	 * : Answer yes to the confirmation message.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp notification delete 520
+	 *     Success: Notification deleted.
+	 *
+	 *     $ wp bp notification delete 55654 54564 --yes
+	 *     Success: Notification deleted.
+	 *
+	 * @alias trash
+	 */
+	public function delete( $args, $assoc_args ) {
+		$notification_id = $args[0];
+
+		WP_CLI::confirm( 'Are you sure you want to delete this notification?', $assoc_args );
+
+		parent::_delete( array( $notification_id ), $assoc_args, function( $notification_id ) {
+
+			$notification = bp_notifications_get_notification( $notification_id );
+
+			if ( empty( $notification->id ) ) {
+				WP_CLI::error( 'No notification found by that ID.' );
+			}
+
+			if ( ! is_object( $notification ) ) {
+				WP_CLI::error( 'Could not find the notification.' );
+			}
+
+			if ( \BP_Notifications_Notification::delete( array( 'id' => $notification_id ) ) ) {
+				return array( 'success', 'Notification deleted.' );
+			} else {
+				return array( 'error', 'Could not delete notification.' );
+			}
+		} );
+	}
+
+	/**
+	 * Generate random notifications.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--action=<action>]
```

Comment:
In the case of activity items, we don't have defaults. Instead, we generate random values for each iteration. I'd mirror that behavior here. (Side note: it would be nice to be able to pass various parameters to `generate` - `--component=groups` etc - and have them override the random behavior. This would be a good future enhancement here and in Activity; for now, I'd make them consistent.)